src/library/scala/collection/immutable/ListSet.scala

```diff
@@ -174,9 +170,17 @@ sealed class ListSet[A] extends AbstractSet[A]
 
     /** `-` can be used to remove a single element from a set.
      */
-    override def -(e: A): ListSet[A] = if (e == head) self else {
-      val tail = self - e; new tail.Node(head)
-    }
+    override def -(e: A): ListSet[A] = removeInternal(e, this, Nil)
+
+    @tailrec private def removeInternal(k: A, cur: ListSet[A], acc: List[ListSet[A]]): ListSet[A] =
+      if (cur.isEmpty)
+        acc.last
+      else if (k == cur.head)
+        (cur.tail /: acc) {
```

Comment:
I disagree.  I find the `/:` form much clearer, as it keeps the accumulated value always-on-the-left.  A high level of expertise can be assumed when reading or writing library code, so I think it's fine.  (Note that the style guide is not definitive.)